src/types/index.ts

```diff
@@ -337,13 +338,21 @@ export interface ClaimScope {
 }
 
 export interface TrustedClaimIssuer {
-  identity: string | Identity;
+  identity: Identity;
   /**
    * an undefined value means that the issuer is trusted for all claim types.
    */
   trustedFor?: ClaimType[];
 }
 
+export type InputTrustedClaimIssuer = Modify<
+  TrustedClaimIssuer,
+  {
+    identity: string | Identity;
+    trustedFor?: ClaimType[];
```

Comment:
This isn't necessary

`Modify<T, R>` creates a type which has the properties of both `T` and `R`, and in case of an overlap, `R` takes precedence. So `Modify<TrustedClaimIssuer, { identity: string | Identity }>` will result in 

```
{
    identity: string | Identity;
    trustedFor?: ClaimType[];
}
```

Without the need to explicitly add `trustedFor?: ClaimType[];`